### PR TITLE
K8s client

### PR DESCRIPTION
**Pull request by rapmetal**
add kubernetes client for api server.
add job_buidler for build a training task.

### CODE REVIEW

File: web_console/constants/index.js [hunk 1]
```diff
@@ -1,6 +1,9 @@
 const DEFAULT_SERVER_CONFIG = {
   K8S_HOST: '127.0.0.1',
   K8S_PORT: 8000,
+  API_VERSION: 'fedlearner.k8s.io/v1alpha1',
+  KIND: 'FLApp',
+  NAMESPACE: 'default',
```
Comment (marswong) [inline web_console/constants/index.js:6]: these params would be better to be designed as the `constructor` params of `KubernetesClient`.

server config is mainly used for process config, mostly the address of remote server

File: web_console/tests/libs/k8s.test.js [hunk 2]
```diff
@@ -11,18 +20,62 @@ describe('Kubernetes Client', () => {
...
+    });
+  });
+  
+  describe('createFLApp', () => {
+    it('should throw for none namespace, job_body', () => {
+      assert.rejects(k8s.createFLApp());
+    });
+
+    it('should create default job for default', async () => {
+      const { flapps } = await k8s.createFLApp('default', loadYaml(testYaml));
```
Comment (marswong) [inline web_console/tests/libs/k8s.test.js:33]: no assert

File: web_console/tests/libs/k8s.test.js [hunk 3]
```diff
@@ -11,18 +20,62 @@ describe('Kubernetes Client', () => {
...
     });
   });
+
+  describe('getFLApp', () => {
+    it('should throw for none namespace, name', () => {
+      assert.rejects(k8s.getFLApp());
+    });
+
+    it('should get all pods for default', async () => {
+      const { flapps } = await k8s.getFLApp('default', 'normal');
```
Comment (marswong) [inline web_console/tests/libs/k8s.test.js:55]: no assert

File: web_console/tests/libs/k8s.test.js [hunk 4]
```diff
@@ -11,18 +20,62 @@ describe('Kubernetes Client', () => {
...
+    });
+  });
+
+  describe('deleteFLApp', () => {
+    it('should throw for none namespace, name', () => {
+      assert.rejects(k8s.deleteFLApp());
+    });
+
+    it('should delete test application for default', async () => {
+      const { flapps } = await k8s.deleteFLApp('default', 'normal');
```
Comment (marswong) [inline web_console/tests/libs/k8s.test.js:76]: no assert

File: web_console/tests/utils/job_builder.test.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
+const assert = require('assert');
```
Comment (marswong) [inline web_console/tests/utils/job_builder.test.js:1]: no assert for the whole test

File: web_console/utils/job_builder.js [hunk 6]
```diff
@@ -0,0 +1,93 @@
+const { dumpYaml } = require("./yaml");
+let server_config;
+try {
+    server_config = require('../server.config');
+} catch (err) {
+    server_config = require('../constants').DEFAULT_SERVER_CONFIG;
+}
+
+function roleConfig(config, public_env={}, public_volume={}) {
+    return config
```
Comment (marswong) [inline web_console/utils/job_builder.js:10]: why return `config` first?

File: web_console/constants/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (rapmetal) [reply web_console/constants/index.js:6]: 以后都没有leader/follower

File: web_console/tests/libs/k8s.test.js [hunk 7]
```diff
@@ -1,5 +1,14 @@
 const assert = require('assert');
+const path = require('path');
+const { loadYaml } = require('../../utils/yaml');
+const { readFileSync } = require('../../utils');
 const KubernetesClient = require('../../libs/k8s');
+const { isObject } = require('util');
```
Comment (marswong) [inline web_console/tests/libs/k8s.test.js:6]: dirty code

**Comment by marswong** [comment]
cc @piiswrong

**Pull request merged**